lib/pbench/server/auth/auth.py

```diff
@@ -101,3 +106,59 @@ def verify_auth(auth_token):
                 e,
             )
         return None
+
+    @staticmethod
+    def verify_third_party_token(
+        auth_token: str, oidc_client: OpenIDClient, tokeninfo_endpoint: str = None
+    ) -> bool:
+        """
+        Verify a token provided to the Pbench server which was obtained from a
+        third party identity provider.
+        Args:
+            auth_token: Token to authenticate
+            oidc_client: OIDC client to call to authenticate the token
+            tokeninfo_endpoint: Optional tokeninfo_endpoint to validate
+                                tokens online in case offline verification
+                                results in some exception.
```

Comment:
tokeninfo endpoint is not provided by all the identity providers for various reasons. So that's why I removed it as an `OpenIDClient` object property. However, now that we are moving towards a direction where the Pbench server would not have to directly talk to any third-party identity provider at all as the point of contact would only be the broker, and the broker is a Keycloak instance, server `OpenIdClient` object would have the `tokeninfo_endpoint`.  

So I guess I should update the `OpenIdClient` class to include the `tokeninfo_endpoint` as a class property and use that here instead of separately passing the parameter.

Reply:
>  the point of contact would only be the broker, and the broker is a Keycloak instance

To the extent that we can, it would be best to avoid any assumptions in the Pbench Server about what the IDP is.  I don't think we (should) care whether it is a broker or a source; it would be best if we didn't care whether it was Keycloak or not (so long as it speaks OIDC); and so on.

So, we expect to be able to receive these endpoints, but we have to be prepared for them to be missing, yes?